### PR TITLE
Rebase and squash #89

### PR DESCRIPTION
**Pull request by indirect**

### CODE REVIEW

**Comment by indirect** [comment]
@homu r+

File: lib/bundler_api/gem_helper.rb [hunk 1]
```diff
@@ -22,16 +22,21 @@ def full_name
...
 
-  def download_spec(base = "http://production.s3.rubygems.org")
-    @mutex.synchronize { return @gemspec if @gemspec }
-    url   = "#{base}/quick/Marshal.4.8/#{full_name}.gemspec.rz"
+  def download_spec(base = nil)
+    base ||= ENV.fetch("DOWNLOAD_BASE", "http://production.s3.rubygems.org")
+    url = "#{base}/quick/Marshal.4.8/#{full_name}.gemspec.rz"
 
     @mutex.synchronize do
-      @gemspec = Marshal.load(Gem.inflate(fetch(url)))
```
Comment (segiddins) [inline lib/bundler_api/gem_helper.rb]: `@gemspec ||= Marshal.load(Gem.inflate(fetch(url)))` ?

Comment (indirect) [inline lib/bundler_api/gem_helper.rb]: lol what a novel idea

**Comment by indirect** [comment]
@homu r+

**Pull request merged**